+siibra/+core/Region.m

```diff
@@ -0,0 +1,102 @@
+classdef Region < handle
+    properties
+        name
+        id
+        parcellation
+        spaceAndRegionUrl
+        spaces
+        parent
+        children
+    end
+    methods
+        function region = Region(name, id, parcellation, dataset_specs)
+            region.name = name;
+            region.id = id;
+            region.parcellation = parcellation;
+            
+            spaceAndRegion.spaces = siibra.core.Space.empty;
+            spaceAndRegion.url = string.empty;
+            space_ids = string.empty;
+            space_urls = string.empty;
+
+            % parse dataset_specs for this region
+            if ~isempty(dataset_specs)
+                for i = 1:numel(dataset_specs)
+                    if iscell(dataset_specs) && isfield(dataset_specs{i, 1}, "space_id")
+                        space_ids(end +1) = dataset_specs{i, 1}.space_id;
+                        space_urls(end +1) = dataset_specs{i, 1}.url;
+                    elseif isfield(dataset_specs(i), "space_id")
+                        space_ids(end +1) = dataset_specs(i).space_id;
+                        space_urls(end +1) = dataset_specs(i).url;
+                    end
+                end
+            end
+            for region_index = 1:numel(space_ids)
+                for parcellation_index = 1:numel(parcellation.spaces)
+                    if space_ids(region_index) == parcellation.spaces(parcellation_index).id
+                        spaceAndRegion.spaces(end + 1) = parcellation.spaces(parcellation_index);
+                        spaceAndRegion.url(end + 1) = space_urls(region_index);
+                        break
+                    end
+                end
+            end
+            region.spaceAndRegionUrl = spaceAndRegion;
+        end
+
+        function spaces = get.spaces(obj)
```

Comment:
Does the get.xxx(obj)  syntax make this appear as an object property instead of  member function?

Reply:
Exactly, you got it :)